Nos/Views/ThreadView.swift

```diff
@@ -20,48 +29,103 @@ struct ThreadView: View {
-                                .fill(Color.secondaryTxt)
-                            }
-                            NoteButton(note: event)
-                                .padding(.horizontal)
-                        }
-                        .onAppear {
-                            // Error scenario: we have an event in core data without an author
-                            guard let author = event.author else {
-                                print("Event author is nil")
-                                return
+                                NoteButton(note: event)
+                                    .padding(.horizontal)
                             }
-                            
-                            if !author.isPopulated {
-                                print("Need to sync author: \(author.hexadecimalPublicKey ?? "")")
-                                authorsToSync.append(author)
+                            .onAppear {
+                                // Error scenario: we have an event in core data without an author
+                                guard let author = event.author else {
+                                    print("Event author is nil")
+                                    return
+                                }
+                                
+                                if !author.isPopulated {
+                                    print("Need to sync author: \(author.hexadecimalPublicKey ?? "")")
+                                    authorsToSync.append(author)
+                                }
                             }
                         }
                     }
                 }
             }
+            .padding(.top, 1)
+            .navigationBarBackButtonHidden(true)
+            .onAppear {
+                router.navigationTitle = Localized.threadView.rawValue
+            }
+            VStack {
+                Spacer()
+                VStack {
+                    ExpandingTextFieldAndSubmitButton( placeholder: "Post a reply", reply: $reply) {
+                        postReply(reply)
+                    }
+                    .padding(.horizontal)
+                    .padding(.vertical, 10)
+                }
+                .background(Color.white)
+            }
         }
-        .padding(.top, 1)
-        .navigationBarBackButtonHidden(true)
-        .onAppear {
-            router.navigationTitle = Localized.threadView.rawValue
+    }
+    
+    func postReply(_ replyText: String) {
+        do {
+            guard let keyPair else {
+                alert = AlertState(title: {
+                    TextState(Localized.error.string)
+                }, message: {
+                    TextState(Localized.youNeedToEnterAPrivateKeyBeforePosting.string)
+                })
+                return
+            }
+            let tags: [[String]] = [["p", note.author!.publicKey!.hex], ["e", note.identifier!]]
```

Comment:
Hey, I think we'll need to add a bit more logic for the e tags here. We'll need an e tag for the root event of the thread, which may not be `note` if we're in a nested thread, but we can get it by looking at `note.eventReferences`. Then, if we're replying to a reply to root, we'll need a second e tag for the replies event id. Both of these e tags will need the proper format too, which means they'll need a recommended relay URL (which we can probably leave as an empty string for now) and a marker (either "root" or "reply"). So the full e tag looks like `["e", {event ID}, {recommended relay URL}, {"root" or "reply"}]`.

We can talk about this offline if you want

Reply:
Thanks for your help!